File: spark/src/main/scala/geotrellis/spark/io/index/RowMajorKeyIndexMethod.scala
package geotrellis.spark.io.index

import geotrellis.spark._
import geotrellis.spark.io.index.rowmajor._

import com.github.nscala_time.time.Imports._

private[index] trait RowMajorKeyIndexMethod

object RowMajorKeyIndexMethod extends RowMajorKeyIndexMethod {
  implicit def spatialKeyIndexMethod(m: RowMajorKeyIndexMethod): KeyIndexMethod[GridKey] =
    new KeyIndexMethod[GridKey] {
      def createIndex(keyBounds: KeyBounds[GridKey]): KeyIndex[GridKey] =
        new RowMajorGridKeyIndex(keyBounds)
    }
}

File: geotrellis-python/geotrellis/spark/io/index/RowMajorKeyIndexMethod.py
from geotrellis.spark.io.index.KeyIndexMethod import KeyIndexMethod
from geotrellis.spark.SpatialKey import SpatialKey
from geotrellis.spark.io.index.rowmajor.RowMajorSpatialKeyIndex import RowMajorSpatialKeyIndex

class RowMajorKeyIndexMethod(object):
    @staticmethod
    def spatialKeyIndexMethod():
        class TempKeyIndexMethod(KeyIndexMethod[SpatialKey]):
            def createIndex(self, keyBounds):
                return RowMajorSpatialKeyIndex(keyBounds)
        return TempKeyIndexMethod()
